docs/reference/src/main/asciidoc/ri-spi.asciidoc

```diff
@@ -979,3 +979,101 @@ Instead, this dependency needs to be deployed separately to the OSGi container.
+
+===== Shaded JARs name change
+
+Weld 2.x created shaded JARs for SE and Servlet with following artifact IDs: `weld-se`, `weld-servlet`.
+In Weld 3.0 a `-shaded` affix was added to the name of these artifacts to make it clear what they contain.
+E.g. new artifact IDs of these JARs are `weld-se-shaded` and `weld-servlet-shaded` respectively.
+
+===== Weld 3 modules
+
+Weld 3.0 was fine-grained and several optional modules were separated from the core.
+Namely, Weld now has Web, JTA, JSF and EJB modules.
+
+The coordinates for these artifacts are as follows:
+
+* WEB - `org.jboss.weld.module:weld-web`
+* JTA - `org.jboss.weld.module:weld-jta`
+* JSF - `org.jboss.weld.module:weld-jsf` 
+* EJB - `org.jboss.weld.module:weld-ejb`
+
+Each module represents the logic behind Weld integration with the given technology.
+
+===== `beans.xml` change for vetoed alternatives
+
+Based on the decision taken in https://issues.jboss.org/browse/CDI-627[CDI-627], the behaviour of vetoed alternatives had changed slightly.
+`beans.xml` validation is now less strict and you can have a vetoed alternative while still having it enabled via `beans.xml`.
+This behaviour is the same as it was in CDI 1.0.
+
+===== `BeanManager#fireEvent` was deprecated
+
+While `BeanManager#fireEvent` still works, users are encouraged to use newly added `BeanManager#getEvent` method instead.
+This method returns an instance of `Event<Object>` built-in bean which can then have the types and qualifiers further specified via `select` method.
+It also allows to fire both kinds of events; synchronous and asynchronous.
+See the example below:
+
+===== Weld-core artifact change 
+
+For historical reasons, Weld 2.x core is released in two versions which *are identical* in their contents.
+Those are `org.jboss.weld:weld-core` and `org.jboss.weld:weld-core-impl`.
+Starting with Weld 3.0, only `org.jboss.weld:weld-core-impl` artifact will be released.
+
+[source.JAVA, java]
+--------------------------------------------------------------------------------------------------
+@Inject
+BeanManager bm;
+
+public void fireMyEvent() {
+    bm.getEvent().select(Any.Literal.INSTANCE).fire(new Payload());
+    bm.getEvent().select(PaymentCheck.class, Default.Literal.INSTANCE)
+        .fireAsync(new Payload());
+}
+--------------------------------------------------------------------------------------------------
+
+===== SPI changes
+
+The SPI changes are mainly important for integrators, such as WildFly.
+Here is a list of changes based on what SPI class/method they affect:
+
+* http://docs.jboss.org/weld/javadoc/3.0/weld-spi/org/jboss/weld/bootstrap/spi/BeanDeploymentArchive.html[`org.jboss.weld.bootstrap.spi.BeanDeploymentArchive`]
+
+** Two new methods - `getKnownClasses()` and `getLoadedBeanClasses()`
+** Both methods have default implementations
+** Integrators might want to override them if they want to make use of preloaded classes during bootstrap
```

Comment:
This only applies to `getLoadedBeanClasses()`. `getKnownClasses()` has a different purpose. See also WELD-2045.